_api-reference/ingest-apis/geoip.md

```diff
@@ -0,0 +1,172 @@
+
+#### Example: PUT request
+
+```json
+{
+    "endpoint": https://geoip.maps.opensearch.org/v1/geolite2-city/manifest.json
+    "update_interval_in_days": 1
+}
+```
+
+#### Example: Response
+
+```json
+{
+    "acknowledged":true
+}
+```
+
+## Creating the IP2geo processor
+
+Once the data source is created, you can create the `Ip2geo` processor. To create the processor, send a PUT request.
+
+#### Example: PUT request
+
+```json
+{
+   "description":"convert ip to geo",
+   "processors":[
+    {
+        "ip2geo":{
+            "field":"_ip",
+            "datsource"::"test1"
+        }
+    }
+   ] 
+}
+
+
+#### Example: Response
+
+```json
+<insert-example-response>
+```
+
+## Using the IP2geo processor in a pipeline
+
+The following table describes <what>.
+
+| Name | Required | Default | Description |
+|------|----------|---------|-------------|
+| field | yes | - | The field to get the ip address from for the geographical lookup. |
+| target_field | no | ip2geo | The field that will hold the geographical information looked up from the Maxmind database. |
+| database | no | geoip.maps.opensearch.org/v1/geolite2-city/manifest.json | The database filename referring to a database the module ships with or a custom database in the ingest-geoip config directory. |
+| properties | no |  Default field depends on what is available in `database`. | Controls what properties are added to the target_field based on the geoip lookup. |
+| ignore_missing | no | false | If `true` and `field` does not exist, the processor quietly exits without modifying the document. |
+| first_only | no | true | If `true` only first found geoip data will be returned, even if field contains array. |
+
+
+The following code is an example of using <example_name> and adds the geographical information to the `geoip` field based on the `ip` field.
+
+### Simulating the pipeline
+
+To simulate the pipeline, specify the pipeline in the POST request.
+
+#### Example: POST request
+
+```json
+<insert-example-request>
+```
+
+## Deleting the Ip2geo processor
+
+To delete the `IP2geo` processor, send a DELETE request. 
+
+#### Example: DELETE request
+
+```json
+{
+    DELETE https://<host>:<port>/_ingest/pipeline/<processor>
+}
+```
+## Deleting the data source
+
+To delete the data source, send a DELETE request. Note that if you have another processors that uses the data source, the delete fails. To delete the dat source, you must delete all processors associated with the data source. 
+
+#### Example: DELETE request
+
+```json
+{
+    DELETE https://<host>:<port>/_plugins/geospatial/ip2geo/datasource/_all
+}
+```
+
+#### Example: Failed DELETE request
+
+```json
+<insert-example-failed-request>
+```
+
+## Next steps
+
+<Do we want to link to any Data Prepper processor information?>
```

Comment:
### Node Settings
The IP2geo datasource and IP2geo processor has node settings

IP2Geo datasource:

* plugins.geospatial.ip2geo.datasource.endpoint
Default endpoint for create datasource API. Defaults to https://geoip.maps.opensearch.org/v1/geolite2-city/manifest.json
* plugins.geospatial.ip2geo.datasource.update_interval_in_days
Default update interval for create datasource API. Defaults to 3.
* plugins.geospatial.ip2geo.datasource.indexing_bulk_size
Max number of documents to ingest in [bulk request](https://opensearch.org/docs/1.2/opensearch/rest-api/document-apis/bulk/) during IP2geo datasource creation process. 

IP2Geo processor:

* plugins.geospatial.ip2geo.processor.max_bundle_size
Maximum number of searches to include in a [multi-search request](https://opensearch.org/docs/latest/api-reference/multi-search/) when enriching documents. Defaults to 100.
* plugins.geospatial.ip2geo.processor.max_concurrent_searches
Maximum number of concurrent [multi-search requests](https://opensearch.org/docs/latest/api-reference/multi-search/) to run when enriching documents. The default depends on your node count and search thread pool size. Higher values can improve performance, but risk overloading the cluster.